core/src/main/java/org/fao/geonet/util/FileUtil.java

```diff
@@ -80,4 +80,14 @@ public static String readLastLines(File file, int lines) {
                 }
         }
     }
+
+    public static String humanizeFileSize(long bytes) {
```

Comment:
Check if could be used: https://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/FileUtils.html#byteCountToDisplaySize(long). 

But I see that it's implemented in Javascript a similar function, so probably good to keep them aligned.

The calculation doesn't seem accurate, the original exception message is `Maximum upload size of 100000000 bytes exceeded`


![error](https://github.com/user-attachments/assets/feea4fef-38bb-460f-94d2-f41a7569045d)


Also the message looks not good: `Payload Too LargeMaximum upload size...`

Reply:
I added javadocs explaining that this was mostly picked due to the extra decimal places.

If we don't want to support new code and prefer to round to the nearest MB then I'm fine with removing this and using the apache commons version.

Regarding the error message, this seems to be caused by the data.errorThrown being prepended to the error message.  Since we translate the other messages we probably don't want to add non translated and translated text at the same time so I moved the assignment of data.errorThrown at the end only in cases where the message was still empty.  This should soft your issue.